libraries/core_libs/consensus/src/dag/dag.cpp

```diff
@@ -610,6 +614,7 @@ uint DagManager::setDagBlockOrder(blk_hash_t const &new_anchor, uint64_t period,
         if (validateBlockNotExpired(dag_block, expired_dag_blocks_to_remove)) {
           addToDag(blk_hash, pivot_hash, dag_block->getTips(), dag_block->getLevel(), false);
         } else {
+          db_->removeDagBlock(blk_hash);
```

Comment:
Dose this also need to remove from seen_blocks_?

Reply:
I don't think so. Seen blocks will make us ignore an expired block if we receive it again.